### PR TITLE
HPCC-17130 Mine button to return owner results

### PR DESCRIPTION
**Pull request by miguelvazq**
Add a owner filter checkbox called "Mine" that will allow that user to
return the currently logged on user's items for specific:
- Workunits
- Files
- DFU Workunits
- Published Queries

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

**Comment by miguelvazq** [comment]
@GordonSmith Rebased.

File: esp/src/eclwatch/DFUQueryWidget.js [hunk 1]
```diff
@@ -449,6 +462,27 @@ define([
...
+                    }
+                }
+            });
+
+            WsAccount.MyAccount({
+            }).then(function (response) {
+                if (lang.exists("MyAccountResponse.username", response)) {
+                    context.userName = response.MyAccountResponse.username;
+                }
+            });
```
Comment (GordonSmith) [inline esp/src/eclwatch/DFUQueryWidget.js]: I think you can get the global user id from dojoConfig?

**Review by GordonSmith — APPROVED** [review]
Looks ok - it may be better to use the global userID from dojoConfig?

**Comment by richardkchapman** [comment]
@miguelvazq

**Comment by miguelvazq** [comment]
@tonymkirk Any feedback on mine button placement? Does it make more sense to have it globally and affect all components vs. individual list views?

**Comment by tonymkirk** [comment]
I could see a global "mine" switch being useful, perhaps, but not in lieu of individual list views.

**Comment by richardkchapman** [comment]
@miguelvazq Needs rebasing for a clean merge

**Comment by miguelvazq** [comment]
Will rebase and move the username to a dojoConfig based call.

**Comment by miguelvazq** [comment]
@GordonSmith Rebased and added dojoConfig

**Review by GordonSmith — APPROVED** [review]
Looks good.

**Pull request merged**